### ISSUE TITLE
Patching Boost 1.73 Fails

### ISSUE THREAD

**Issue by geoffthemedio** [freeorion/freeorion-sdk#68]
I wanted to be able to build against Boost 1.73 to try to fix some reported FreeOrion build errors, so I edited `Sources.cmake` to build Boost 1.73 (with the MD5 from the SourceForge download link):
````
set(BOOST_VERSION 1.73.0)
string(REPLACE "." "_" BOOST_VERSION_ ${BOOST_VERSION})
set(BOOST_URL http://downloads.sourceforge.net/project/boost/boost/${BOOST_VERSION}/boost_${BOOST_VERSION_}.tar.gz)
set(BOOST_MD5 4036cd27ef7548b8d29c30ea10956196)
````

When I try to build the SDK, I get an error
````
patching file boost/gil/channel_algorithm.hpp
  Hunk #1 FAILED at 51.
````

More of the log for context:
````
Done Building Project "C:\Users\Geoff\Desktop\freeorion-sdk-boost173\build\python3.vcxproj" (default targets).
PrepareForBuild:
  Creating directory "Win32\RelWithDebInfo\boost\".
  Creating directory "Win32\RelWithDebInfo\boost\boost.tlog\".
InitializeBuildStatus:
  Creating "Win32\RelWithDebInfo\boost\boost.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
ComputeCustomBuildOutput:
  Creating directory "C:\Users\Geoff\Desktop\freeorion-sdk-boost173\build\boost-prefix\src\boost-stamp\RelWithDebInfo\".
CustomBuild:
  Creating directories for 'boost'
  Performing download step (download, verify and extract) for 'boost'
  -- Downloading...
     dst='C:/Users/Geoff/Desktop/freeorion-sdk-boost173/build/downloads/boost_1_73_0.tar.gz'
     timeout='none'
  -- Using src='http://downloads.sourceforge.net/project/boost/boost/1.73.0/boost_1_73_0.tar.gz'
  -- [download 100% complete]
  -- [download 0% complete]
  -- [download 1% complete]
  -- [download 2% complete]
 [...]
  -- [download 100% complete]
  -- verifying file...
         file='C:/Users/Geoff/Desktop/freeorion-sdk-boost173/build/downloads/boost_1_73_0.tar.gz'
  -- Downloading... done
  -- extracting...
       src='C:/Users/Geoff/Desktop/freeorion-sdk-boost173/build/downloads/boost_1_73_0.tar.gz'
       dst='C:/Users/Geoff/Desktop/freeorion-sdk-boost173/build/boost-prefix/src/boost'
  -- extracting... [tar xfz]
  -- extracting... [analysis]
  -- extracting... [rename]
  -- extracting... [clean up]
  -- extracting... done
  No update step for 'boost'
  Performing patch step for 'boost'
  patching file boost/log/detail/setup_config.hpp
  patching file tools/build/src/tools/package.jam
  Hunk #1 succeeded at 234 (offset 135 lines).
  patching file tools/build/src/tools/darwin.jam
  Hunk #2 succeeded at 201 (offset -3 lines).
  patching file boost/gil/channel_algorithm.hpp
  Hunk #1 FAILED at 51.
  1 out of 1 hunk FAILED -- saving rejects to file boost/gil/channel_algorithm.hpp.rej
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(209,5): error MSB6006: "cmd.exe" exited with code 1. [C:\Users\Geoff\Desktop\freeorion-sdk-boost173\build\boost.vcxproj]
Done Building Project "C:\Users\Geoff\Desktop\freeorion-sdk-boost173\build\boost.vcxproj" (default targets) -- FAILED.
````

**Comment by o01eg** [comment]
Could you try to base on #54?

**Comment by geoffthemedio** [comment]
> Could you try to base on #54?

That built the SDK OK.

**Comment by o01eg** [comment]
Is MacOS issue fixed in 1.73?

**Comment by Vezzra** [comment]
@o01eg, which macOS issue are you referring to?

**Comment by o01eg** [comment]
@Vezzra CI broken on MacOS when I tried to update boost.

**Comment by geoffthemedio** [comment]
> That built the SDK OK.

FreeOrion also builds and runs OK with the SDK using Boost 1.73

**Comment by o01eg** [comment]
I suppose it is fixed now with updating to boost 1.74 in https://github.com/freeorion/freeorion-sdk/pull/83
